src/test/ui/enum/union-in-enum.rs

```diff
@@ -0,0 +1,5 @@
+enum A { union } //~ WARNING
```

Comment:
Adding `// check-pass` would be good here, since we don't need to test codegen and linking for this (and we expect the test to pass, not fail, which would fix the current CI failure).